ui/core/src/api/SifService/SifService.ts

```diff
@@ -117,6 +124,37 @@ export default function createSifService({
 
     async connect() {
       // connect to Keplr
+      console.log('connect service', keplrChainConfig, keplrProvider)
+      if (!keplrProvider) {
+        throw {message: "Keplr Not Found", detail: "Check if extension enabled for this URL"}
+      }
```

Comment:
We probably don't want to do this. We can merge it and change later if you like but we should really pull all notification code out of the service layer.